### PR TITLE
feat(vscode_projects): add vscode project plugin

### PR DESCRIPTION
**Pull request by cathaysia**
Signed-off-by: Longtao Zhang <DragonBillow@outlook.com>

### CODE REVIEW

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:2]: this string appears in the settings and should serve as longer description of the plugin

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:13]: has there been a 1.0?

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:15]: This is a human readable pretty printed string. i should be brief and imperative like "Open VSCode projects"

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:14]: This is a human readable pretty printed string. Use VSCode projects

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:17]: no not kill

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:21]: try to generalize this. this will work on linux only.

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:30]: use md_id here

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:38]: drop if unused

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:68]: ?! this should be usable for the entire worldwide community

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:69]: same here

**Review by ManuelSchneid3r — CHANGES_REQUESTED** [review]
This plugin is far from polished. Being in this unfinished state, why do you send this PR at all? Do you really expect me to merge and ship it? Please polish, otherwise i cant merge it.

**Comment by cathaysia** [comment]
OK, I give some changes on codes

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:30]: iid is the interface identifier. Use md_id

**Review by ManuelSchneid3r — CHANGES_REQUESTED** [review]

File: vscode_projects/__init__.py
Comment (ManuelSchneid3r) [inline vscode_projects/__init__.py:64]: this should be the identifier of a project item right? identifiers have to be unique.

**Review by ManuelSchneid3r — COMMENTED** [review]
I think you got that wrong. The identifier has to be unique such that each item has its own idententifier, use project ids or path or the name of the project (but actually this is probably not guaranteed to be unique as well)

**Comment by cathaysia** [comment]
> I think you got that wrong. The identifier has to be unique such that each item has its own idententifier, use project ids or path or the name of the project (but actually this is probably not guaranteed to be unique as well)

can I use uuid.uuid1() ?

**Comment by ManuelSchneid3r** [comment]
@cathaysia why did you close the pr? 

> can I use uuid.uuid1() ?

why not just sth like `'md_id' + project_path`?

The id has to be persistant (because it will be reused for sorting) and unique (because it is used as icon cache key).

**Comment by cathaysia** [comment]
I worry that I don't have much time to maintain this code. This plugin was originally written just because I needed this plugin. If someone is happy to maintain it, you can simply copy and modify my code.

**Comment by ManuelSchneid3r** [comment]
You dont necessarily have to maintain it. Its fine to leave the maintainer field empty, but initially the plugin should be polished before its shipped around the world.

**Pull request closed without merging**
